hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -344,14 +344,27 @@ private StoreContext initializeStoreContext(ColumnFamilyDescriptor family) throw
   }
 
   private InetSocketAddress[] getFavoredNodes() {
-    InetSocketAddress[] favoredNodes = null;
     if (region.getRegionServerServices() != null) {
-      favoredNodes = region.getRegionServerServices().getFavoredNodesForRegion(
-          region.getRegionInfo().getEncodedName());
+      return region.getRegionServerServices()
+          .getFavoredNodesForRegion(region.getRegionInfo().getEncodedName());
     }
     return favoredNodes;
   }
 
+  // Favored nodes used by compaction offload
+  private InetSocketAddress[] favoredNodes = null;
```

Comment:
OK we added a class field here... It is calculated by a public method? Will call be called multiple times? Thread safe?

Reply:
Not thread safe, but only called once after `initializeStoreContext` and before real do compaction.For we initialize a new store everytime for a compaction request.

Reply:
Is it possible to not make this method public? Or at least change the method name? The name setXXX seems that we can call it multiple times...

Reply:
Must be public for not same package, Let me change name to 'assignFavoredNodesForCompactionOffload' and add annotate